api/provider/aws/task_list.go

```diff
@@ -0,0 +1,151 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/config"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+const MAX_TASK_IDS = 100
+
+func (t *TaskProvider) List() ([]models.TaskSummary, error) {
+	clusterNames, err := t.listClusterNames()
+	if err != nil {
+		return nil, err
+	}
+
+	taskCopies := map[string][]*ecs.Task{}
+	for _, clusterName := range clusterNames {
+		environmentID := delLayer0Prefix(t.Config.Instance(), clusterName)
+
+		running := "RUNNING"
+		tasks, err := t.listTasksByStatus(&running, &environmentID)
+		if err != nil {
+			return nil, err
+		}
+
+		stopped := "STOPPED"
+		stoppedTasks, err := t.listTasksByStatus(&stopped, &environmentID)
+		if err != nil {
+			return nil, err
+		}
+
+		taskARNs := append(tasks, stoppedTasks...)
+		if len(taskARNs) > 0 {
+			describeTasksInput := &ecs.DescribeTasksInput{}
+			describeTasksInput.SetCluster(clusterName)
+
+			tasks, err := t.describeTasks(describeTasksInput)
+
+			if err != nil {
+				return nil, err
+			}
+
+			for _, task := range tasks {
+				if task.StartedBy != nil {
+					if strings.HasPrefix(*task.StartedBy, config.ENVVAR_INSTANCE) {
+						if _, ok := taskCopies[*task.StartedBy]; !ok {
+							taskCopies[*task.StartedBy] = []*ecs.Task{}
+						}
+
+						taskCopies[*task.StartedBy] = append(taskCopies[*task.StartedBy], task)
+					}
+				}
+			}
+		}
+	}
+
+	getModel := func(tasks []*ecs.Task) (*models.Task, error) {
+		if len(tasks) == 0 {
+			return nil, errors.Newf(errors.TaskDoesNotExist, "The specified task does not exist")
+		}
+
+		model := &models.Task{
+			EnvironmentID: strings.SplitN(*tasks[0].ClusterArn, "/", 2)[1],
+			TaskID:        ECSTaskID(*tasks[0].StartedBy).L0TaskID(),
+		}
+
+		return model, nil
+	}
+
+	tasks := []*models.Task{}
+	for i, copies := range taskCopies {
+
+		model, err := getModel(copies)
+		if err != nil {
+			return nil, err
+		}
+
+		tasks = append(tasks, model)
+	}
+
+	summaries := make([]models.TaskSummary, len(tasks))
+	for i, task := range tasks {
+		environmentID := delLayer0Prefix(t.Config.Instance(), clusterName)
+		summary := models.TaskSummary{
+			EnvironmentID: environmentID,
+		}
+
+		summaries[i] = summary
+	}
+
+	if err := t.populateSummariesTags(summaries); err != nil {
+		return nil, err
+	}
+
+	return summaries, nil
+
+}
+
+func (t *TaskProvider) describeTasks(describeTasksInput *ecs.DescribeTasksInput) ([]*ecs.Task, error) {
+	ret := []*ecs.Task{}
+	for i, task := range describeTasksInput.Tasks {
+		if i > MAX_TASK_IDS {
+			i = MAX_TASK_IDS
+		}
+
+		output, err := t.AWS.ECS.DescribeTasks(describeTasksInput)
+		if err != nil {
+			return nil, err
+		}
+
+		ret = append(ret, output.Tasks...)
+	}
+
+	return ret, nil
+
+}
+
+func (t *TaskProvider) listTasksByStatus(status *string, environmentID *string) ([]*string, error) {
```

Comment:
Good choice for a helper function :+1: 

Some nitpick items: 
* arguments should be as simple as possible, so use `string` args and not `*string`.
* follow the existing pattern for making aws api calls ([for example](https://github.com/quintilesims/layer0/blob/api-refactor/api/provider/aws/environment_create.go#L133)). The current pattern is much more readable .